velox/common/caching/AsyncDataCache.cpp

```diff
@@ -32,7 +32,7 @@ AsyncDataCacheEntry::AsyncDataCacheEntry(CacheShard* shard) : shard_(shard) {
 }
 
 AsyncDataCacheEntry::~AsyncDataCacheEntry() {
-  shard_->cache()->freeNonContiguous(data_);
+  shard_->cache()->allocator()->freeNonContiguous(data_);
```

Comment:
Let's provide a allocator() method to cache shard for this access?

Reply:
Probably not needed. It is pretty much this one use case that chains 2 things together.